src/parsing.jl

```diff
@@ -135,15 +142,29 @@ pqparse(::Type{T}, str::AbstractString) where {T} = parse(T, str)
 # allow parsing as a Symbol anything which works as a String
 pqparse(::Type{Symbol}, str::AbstractString) = Symbol(str)
 
+function generate_binary_parser(symbol)
+    @eval function Base.parse(
+        ::Type{T}, pqv::PQBinaryValue{$(oid(symbol))}
+    ) where {T<:Number}
+        return convert(
+            T, ntoh(unsafe_load(Ptr{$(_DEFAULT_TYPE_MAP[symbol])}(data_pointer(pqv))))
+        )
+    end
+end
+
 ## integers
 _DEFAULT_TYPE_MAP[:int2] = Int16
 _DEFAULT_TYPE_MAP[:int4] = Int32
 _DEFAULT_TYPE_MAP[:int8] = Int64
 
+generate_binary_parser.((:int2, :int4, :int8))
```

Comment:
```suggestion
foreach(generate_binary_parser, (:int2, :int4, :int8))
```

Don't use broadcasting if it's not necessary; iteration is good enough here. Same below for floats.